### PR TITLE
Deprecate `Kokkos::Impl::is_view`

### PR DESCRIPTION
**Pull request by dalg24**
Rational:  Seems to only be there for backward compatibility.  We don't want users to reach into `Kokkos::Impl::`.  Let us emit deprecation warnings so we can remove in upcoming releases.

### CODE REVIEW

**Review by PhilMiller — APPROVED** [review]

**Review by crtrott — APPROVED** [review]

**Comment by crtrott** [comment]
Add to deprecation list in the wiki!

**Comment by dalg24** [comment]
> Add to deprecation list in the wiki!

I'd argue it is in `Impl::` so we don't have to give notice.

Also, are you referring to https://github.com/kokkos/kokkos/wiki/DeprecationPage ?  It does not really fit with the rest.

File: core/src/Kokkos_View.hpp [hunk 1]
```diff
@@ -2123,13 +2123,16 @@ KOKKOS_INLINE_FUNCTION DeducedCommonPropsType<Views...> common_view_alloc_prop(
 
 }  // namespace Kokkos
 
+#ifdef KOKKOS_ENABLE_DEPRECATED_CODE_3
 namespace Kokkos {
 namespace Impl {
 
-using Kokkos::is_view;
+template <class T>
+using is_view KOKKOS_DEPRECATED = Kokkos::is_view<T>;
```
Comment (masterleinad) [inline core/src/Kokkos_View.hpp:2131]:
```suggestion
using is_view KOKKOS_DEPRECATED_WITH_COMMENT("Use  Kokkos::is_view instead!") = Kokkos::is_view<T>;
```

**Comment by dalg24** [comment]
Failure on Jenkins (performance) is unrelated.

**Review by masterleinad — APPROVED** [review]

**Comment by crtrott** [comment]
Yeah ok don't need in the deprecation list ..

**Comment by dalg24** [comment]
SYCL failure is unrelated.  (fetnat05 running out of disk space)

**Pull request merged**